### ISSUE TITLE
Encoding job fails after disabling WebTorrent encoding

### ISSUE THREAD

**Issue by violoncelloCH** [Chocobozzz/PeerTube#4227]
**Describe the current behavior**
after uploading a video (generated with kdenlive: `properties=x265-medium f=mp4 vcodec=libx265 acodec=aac crf=24 ab=320+'k'`) the transcoding process fails.
The admin jobs page lists one single transcoding job for the video with the follwing details:

```
Job: 1344 
Type: video-transcoding 
Processed on Jul 4, 2021, 5:53:19 PM 
Finished on Jul 4, 2021, 5:53:52 PM
--
{   
"type": "optimize-to-webtorrent",   
"videoUUID": "REDACTED",   
"isNewVideo": true 
}
Error: ffmpeg was killed with signal SIGKILL
     at ChildProcess.<anonymous> (/app/node_modules/fluent-ffmpeg/lib/processor.js:180:22)
     at ChildProcess.emit (events.js:314:20)
     at ChildProcess.EventEmitter.emit (domain.js:483:12)
     at Process.ChildProcess._handle.onexit (internal/child_process.js:276:12)
```
(see detailed server error log below)
The video stays unpublished with the message `The video is being transcoded, it may not work properly.` .
After manually triggering a transcoding via cli, the correct hls jobs are created:
```
$ NODE_CONFIG_DIR=/config NODE_ENV=production npm run create-transcoding-job -- -v REDACTED

> peertube@3.2.1 create-transcoding-job /app
> node ./dist/scripts/create-transcoding-job.js "-v" "REDACTED"

Transcoding job for video REDACTED created.
Transcoding job for video REDACTED created.
Transcoding job for video REDACTED created.
Transcoding job for video REDACTED created.
Transcoding job for video REDACTED created.
Transcoding job for video REDACTED created.
Transcoding job for video REDACTED created.
```
The jobs page now lists `new-resolution-to-hls` jobs for the video for each resolution.

**Steps to reproduce:**

1. have a Peertube server with both WebTorrent and HLS encoding enabled
2. update to v3.2
3. disable WebTorrent encoding in the VOD settings
4. upload a video and see the transcoding job failing

**Describe the expected behavior**
encoding should run for HLS and not fail

**Additional information**

* PeerTube instance:
  * URL: video.violoncello.ch
  * version: 3.2.1 / latest docker
  * NodeJS version: as in latest docker image
  * Ffmpeg version: as in latest docker image

<details>
<summary>related server logs</summary>

```
error[7/4/2021, 5:53:52 PM] Error in ffmpeg.

{
  "stdout": "",
  "stderr": "ffmpeg version 4.1.6-1~deb10u1 Copyright (c) 2000-2020 the FFmpeg developers\n  built with gcc 8 (Debian 8.3.0-6)\n  configuration: --prefix=/usr --extra-version='1~deb10u1' --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-avresample --disable-filter=resample --enable-avisynth --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librsvg --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opengl --enable-sdl2 --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --enable-libx264 --enable-shared\n  libavutil      56. 22.100 / 56. 22.100\n  libavcodec     58. 35.100 / 58. 35.100\n  libavformat    58. 20.100 / 58. 20.100\n  libavdevice    58.  5.100 / 58.  5.100\n  libavfilter     7. 40.101 /  7. 40.101\n  libavresample   4.  0.  0 /  4.  0.  0\n  libswscale      5.  3.100 /  5.  3.100\n  libswresample   3.  3.100 /  3.  3.100\n  libpostproc    55.  3.100 / 55.  3.100\nInput #0, mov,mp4,m4a,3gp,3g2,mj2, from '/data/videos/REDACTED-2160.mp4':\n  Metadata:\n    major_brand     : isom\n    minor_version   : 512\n    compatible_brands: isomiso2mp41\n    encoder         : Lavf58.76.100\n  Duration: 00:23:13.88, start: 0.000000, bitrate: 22830 kb/s\n    Stream #0:0(und): Video: hevc (Main) (hev1 / 0x31766568), yuv420p(tv, bt709, progressive), 3840x2160 [SAR 1:1 DAR 16:9], 22504 kb/s, 50 fps, 50 tbr, 12800 tbn, 50 tbc (default)\n    Metadata:\n      handler_name    : VideoHandler\n    Stream #0:1(und): Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, stereo, fltp, 316 kb/s (default)\n    Metadata:\n      handler_name    : SoundHandler\nStream mapping:\n  Stream #0:0 -> #0:0 (hevc (native) -> h264 (libx264))\n  Stream #0:1 -> #0:1 (copy)\nPress [q] to stop, [?] for help\n[libx264 @ 0x55e77bd5f3c0] using SAR=1/1\n[libx264 @ 0x55e77bd5f3c0] frame MB size (240x135) > level limit (3600)\n[libx264 @ 0x55e77bd5f3c0] DPB size (4 frames, 129600 mbs) > level limit (0 frames, 18000 mbs)\n[libx264 @ 0x55e77bd5f3c0] VBV bitrate (22830) > level limit (17500)\n[libx264 @ 0x55e77bd5f3c0] VBV buffer (45661) > level limit (17500)\n[libx264 @ 0x55e77bd5f3c0] MB rate (1620000) > level limit (108000)\n[libx264 @ 0x55e77bd5f3c0] using cpu capabilities: MMX2 SSE2Slow\n[libx264 @ 0x55e77bd5f3c0] profile High, level 3.1\n[libx264 @ 0x55e77bd5f3c0] 264 - core 155 r2917 0a84d98 - H.264/MPEG-4 AVC codec - Copyleft 2003-2018 - http://www.videolan.org/x264.html - options: cabac=1 ref=1 deblock=1:0:0 analyse=0x3:0x113 me=hex subme=2 psy=1 psy_rd=1.00:0.00 mixed_ref=0 me_range=16 chroma_me=1 trellis=0 8x8dct=1 cqm=0 deadzone=21,11 fast_pskip=1 chroma_qp_offset=0 threads=4 lookahead_threads=1 sliced_threads=0 nr=0 decimate=1 interlaced=0 bluray_compat=0 constrained_intra=0 bframes=16 b_pyramid=2 b_adapt=1 b_bias=0 direct=1 weightb=1 open_gop=0 weightp=1 keyint=100 keyint_min=10 scenecut=40 intra_refresh=0 rc_lookahead=10 rc=crf mbtree=1 crf=23.0 qcomp=0.60 qpmin=0 qpmax=69 qpstep=4 vbv_maxrate=22830 vbv_bufsize=45661 crf_max=0.0 nal_hrd=none filler=0 ip_ratio=1.40 aq=1:1.00\nOutput #0, mp4, to '/data/tmp/124-transcoded.mp4':\n  Metadata:\n    encoder         : Lavf58.20.100\n    Stream #0:0: Video: h264 (libx264) (avc1 / 0x31637661), yuv420p, 3840x2160 [SAR 1:1 DAR 16:9], q=-1--1, 50 fps, 12800 tbn, 50 tbc (default)\n    Metadata:\n      encoder         : Lavc58.35.100 libx264\n    Side data:\n      cpb: bitrate max/min/avg: 22830000/0/0 buffer size: 45661000 vbv_delay: -1\n    Stream #0:1: Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, stereo, fltp, 316 kb/s (default)\nframe=    1 fps=0.0 q=0.0 size=       0kB time=00:00:00.17 bitrate=   2.2kbits/s speed= 0.3x    \nframe=    4 fps=3.2 q=0.0 size=       0kB time=00:00:00.23 bitrate=   1.6kbits/s speed=0.186x    \nframe=    6 fps=3.0 q=0.0 size=       0kB time=00:00:00.27 bitrate=   1.4kbits/s speed=0.139x    \nframe=    9 fps=3.2 q=0.0 size=       0kB time=00:00:00.29 bitrate=   1.3kbits/s speed=0.107x    \nframe=   10 fps=2.8 q=0.0 size=       0kB time=00:00:00.34 bitrate=   1.1kbits/s speed=0.0958x    \nframe=   12 fps=2.7 q=0.0 size=       0kB time=00:00:00.38 bitrate=   1.0kbits/s speed=0.0875x    \nframe=   14 fps=2.7 q=0.0 size=       0kB time=00:00:00.42 bitrate=   0.9kbits/s speed=0.081x    \nframe=   16 fps=2.6 q=0.0 size=       0kB time=00:00:00.46 bitrate=   0.8kbits/s speed=0.076x    \nframe=   17 fps=2.5 q=0.0 size=       0kB time=00:00:00.49 bitrate=   0.8kbits/s speed=0.0733x    \nframe=   19 fps=2.5 q=0.0 size=       0kB time=00:00:00.53 bitrate=   0.7kbits/s speed=0.0696x    \nframe=   20 fps=2.4 q=0.0 size=       0kB time=00:00:00.55 bitrate=   0.7kbits/s speed=0.0674x    \nframe=   21 fps=2.4 q=0.0 size=       0kB time=00:00:00.57 bitrate=   0.7kbits/s speed=0.0658x    \nframe=   22 fps=2.4 q=0.0 size=       0kB time=00:00:00.59 bitrate=   0.6kbits/s speed=0.0642x    \nframe=   23 fps=2.3 q=0.0 size=       0kB time=00:00:00.61 bitrate=   0.6kbits/s speed=0.0628x    \nframe=   25 fps=2.3 q=0.0 size=       0kB time=00:00:00.64 bitrate=   0.6kbits/s speed=0.0582x    \nframe=   27 fps=2.3 q=0.0 size=       0kB time=00:00:00.68 bitrate=   0.6kbits/s speed=0.0573x    \nframe=   28 fps=2.2 q=0.0 size=       0kB time=00:00:00.70 bitrate=   0.5kbits/s speed=0.0561x    \nframe=   29 fps=2.2 q=0.0 size=       0kB time=00:00:00.72 bitrate=   0.5kbits/s speed=0.055x    \nframe=   30 fps=2.1 q=0.0 size=       0kB time=00:00:00.74 bitrate=   0.5kbits/s speed=0.0527x    \nframe=   33 fps=2.2 q=0.0 size=       0kB time=00:00:00.81 bitrate=   0.5kbits/s speed=0.055x    \nframe=   34 fps=2.1 q=0.0 size=       0kB time=00:00:00.83 bitrate=   0.5kbits/s speed=0.051x    \nframe=   35 fps=1.8 q=0.0 size=       0kB time=00:00:00.85 bitrate=   0.5kbits/s speed=0.045x    \nframe=   36 fps=1.7 q=0.0 size=       0kB time=00:00:00.87 bitrate=   0.4kbits/s speed=0.0419x    \nframe=   37 fps=1.6 q=0.0 size=       0kB time=00:00:00.89 bitrate=   0.4kbits/s speed=0.0378x    \nframe=   38 fps=1.5 q=31.0 size=       0kB time=00:00:00.91 bitrate=   0.4kbits/s speed=0.037x    \nframe=   44 fps=1.7 q=31.0 size=       0kB time=00:00:01.02 bitrate=   0.4kbits/s speed=0.0392x    \n"
}

error[7/4/2021, 5:53:52 PM] Cannot execute job 1344 in queue video-transcoding.

{
  "payload": {
    "type": "optimize-to-webtorrent",
    "videoUUID": "REDACTED",
    "isNewVideo": true
  },
  "err": {
    "stack": "Error: ffmpeg was killed with signal SIGKILL\n    at ChildProcess.<anonymous> (/app/node_modules/fluent-ffmpeg/lib/processor.js:180:22)\n    at ChildProcess.emit (events.js:314:20)\n    at ChildProcess.EventEmitter.emit (domain.js:483:12)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:276:12)",
    "message": "ffmpeg was killed with signal SIGKILL"
  }
}
```
</details>

**Comment by violoncelloCH** [comment]
I now have to add, that the highest resolution (2160) still fails to transcode for hls. This reproduces also if I create a specific encoding job for that resolution again.

<details>
<summary>job / server log</summary>

```
error[7/4/2021, 8:46:25 PM] Error in ffmpeg.

{
  "stdout": "",
  "stderr": "ffmpeg version 4.1.6-1~deb10u1 Copyright (c) 2000-2020 the FFmpeg developers\n  built with gcc 8 (Debian 8.3.0-6)\n  configuration: --prefix=/usr --extra-version='1~deb10u1' --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-avresample --disable-filter=resample --enable-avisynth --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librsvg --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opengl --enable-sdl2 --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --enable-libx264 --enable-shared\n  libavutil      56. 22.100 / 56. 22.100\n  libavcodec     58. 35.100 / 58. 35.100\n  libavformat    58. 20.100 / 58. 20.100\n  libavdevice    58.  5.100 / 58.  5.100\n  libavfilter     7. 40.101 /  7. 40.101\n  libavresample   4.  0.  0 /  4.  0.  0\n  libswscale      5.  3.100 /  5.  3.100\n  libswresample   3.  3.100 /  3.  3.100\n  libpostproc    55.  3.100 / 55.  3.100\nInput #0, mov,mp4,m4a,3gp,3g2,mj2, from '/data/videos/REDACTED-2160.mp4':\n  Metadata:\n    major_brand     : isom\n    minor_version   : 512\n    compatible_brands: isomiso2mp41\n    encoder         : Lavf58.76.100\n  Duration: 00:23:13.88, start: 0.000000, bitrate: 22830 kb/s\n    Stream #0:0(und): Video: hevc (Main) (hev1 / 0x31766568), yuv420p(tv, bt709, progressive), 3840x2160 [SAR 1:1 DAR 16:9], 22504 kb/s, 50 fps, 50 tbr, 12800 tbn, 50 tbc (default)\n    Metadata:\n      handler_name    : VideoHandler\n    Stream #0:1(und): Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, stereo, fltp, 316 kb/s (default)\n    Metadata:\n      handler_name    : SoundHandler\nStream mapping:\n  Stream #0:0 -> #0:0 (hevc (native) -> h264 (libx264))\n  Stream #0:1 -> #0:1 (copy)\nPress [q] to stop, [?] for help\n[libx264 @ 0x5605e0b6f380] using SAR=1/1\n[libx264 @ 0x5605e0b6f380] frame MB size (240x135) > level limit (3600)\n[libx264 @ 0x5605e0b6f380] DPB size (4 frames, 129600 mbs) > level limit (0 frames, 18000 mbs)\n[libx264 @ 0x5605e0b6f380] VBV bitrate (22830) > level limit (17500)\n[libx264 @ 0x5605e0b6f380] VBV buffer (45661) > level limit (17500)\n[libx264 @ 0x5605e0b6f380] MB rate (1620000) > level limit (108000)\n[libx264 @ 0x5605e0b6f380] using cpu capabilities: MMX2 SSE2Slow\n[libx264 @ 0x5605e0b6f380] profile High, level 3.1\n[libx264 @ 0x5605e0b6f380] 264 - core 155 r2917 0a84d98 - H.264/MPEG-4 AVC codec - Copyleft 2003-2018 - http://www.videolan.org/x264.html - options: cabac=1 ref=1 deblock=1:0:0 analyse=0x3:0x113 me=hex subme=2 psy=1 psy_rd=1.00:0.00 mixed_ref=0 me_range=16 chroma_me=1 trellis=0 8x8dct=1 cqm=0 deadzone=21,11 fast_pskip=1 chroma_qp_offset=0 threads=4 lookahead_threads=1 sliced_threads=0 nr=0 decimate=1 interlaced=0 bluray_compat=0 constrained_intra=0 bframes=16 b_pyramid=2 b_adapt=1 b_bias=0 direct=1 weightb=1 open_gop=0 weightp=1 keyint=100 keyint_min=10 scenecut=40 intra_refresh=0 rc_lookahead=10 rc=crf mbtree=1 crf=23.0 qcomp=0.60 qpmin=0 qpmax=69 qpstep=4 vbv_maxrate=22830 vbv_bufsize=45661 crf_max=0.0 nal_hrd=none filler=0 ip_ratio=1.40 aq=1:1.00\n[hls @ 0x5605e0acf480] Opening '/data/tmp/hls/REDACTED-2160-fragmented.mp4' for writing\nOutput #0, hls, to '/data/tmp/hls/2160.m3u8':\n  Metadata:\n    encoder         : Lavf58.20.100\n    Stream #0:0: Video: h264 (libx264), yuv420p, 3840x2160 [SAR 1:1 DAR 16:9], q=-1--1, 50 fps, 12800 tbn, 50 tbc (default)\n    Metadata:\n      encoder         : Lavc58.35.100 libx264\n    Side data:\n      cpb: bitrate max/min/avg: 22830000/0/0 buffer size: 45661000 vbv_delay: -1\n    Stream #0:1: Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, stereo, fltp, 316 kb/s (default)\nframe=    2 fps=0.0 q=0.0 size=N/A time=00:00:00.19 bitrate=N/A speed=0.355x    \nframe=    5 fps=4.2 q=0.0 size=N/A time=00:00:00.25 bitrate=N/A speed=0.215x    \nframe=   10 fps=5.3 q=0.0 size=N/A time=00:00:00.34 bitrate=N/A speed=0.179x    \nframe=   12 fps=4.8 q=0.0 size=N/A time=00:00:00.38 bitrate=N/A speed=0.154x    \nframe=   17 fps=5.6 q=0.0 size=N/A time=00:00:00.49 bitrate=N/A speed=0.162x    \nframe=   20 fps=5.6 q=0.0 size=N/A time=00:00:00.55 bitrate=N/A speed=0.154x    \nframe=   22 fps=5.1 q=0.0 size=N/A time=00:00:00.59 bitrate=N/A speed=0.139x    \nframe=   25 fps=5.0 q=0.0 size=N/A time=00:00:00.64 bitrate=N/A speed=0.128x    \nframe=   27 fps=4.6 q=0.0 size=N/A time=00:00:00.68 bitrate=N/A speed=0.118x    \nframe=   29 fps=4.2 q=0.0 size=N/A time=00:00:00.72 bitrate=N/A speed=0.104x    \nframe=   30 fps=3.7 q=0.0 size=N/A time=00:00:00.74 bitrate=N/A speed=0.0917x    \nframe=   31 fps=3.2 q=0.0 size=N/A time=00:00:00.76 bitrate=N/A speed=0.0803x    \nframe=   32 fps=2.9 q=0.0 size=N/A time=00:00:00.78 bitrate=N/A speed=0.0707x    \nframe=   33 fps=2.7 q=0.0 size=N/A time=00:00:00.81 bitrate=N/A speed=0.0658x    \nframe=   36 fps=2.4 q=0.0 size=N/A time=00:00:00.87 bitrate=N/A speed=0.058x    \nframe=   37 fps=2.2 q=0.0 size=N/A time=00:00:00.89 bitrate=N/A speed=0.054x    \nframe=   38 fps=1.7 q=31.0 size=N/A time=00:00:00.91 bitrate=N/A speed=0.0408x    \nframe=   39 fps=1.7 q=31.0 size=N/A time=00:00:00.93 bitrate=N/A speed=0.04x    \nframe=   41 fps=1.7 q=31.0 size=N/A time=00:00:00.96 bitrate=N/A speed=0.0388x    \nframe=   43 fps=1.7 q=31.0 size=N/A time=00:00:01.00 bitrate=N/A speed=0.0394x    \n"
}

error[7/4/2021, 8:46:25 PM] Cannot execute job 1351 in queue video-transcoding.

{
  "payload": {
    "type": "new-resolution-to-hls",
    "videoUUID": "REDACTED",
    "resolution": 2160,
    "isPortraitMode": false,
    "copyCodecs": false,
    "isMaxQuality": false
  },
  "err": {
    "stack": "Error: ffmpeg was killed with signal SIGKILL\n    at ChildProcess.<anonymous> (/app/node_modules/fluent-ffmpeg/lib/processor.js:180:22)\n    at ChildProcess.emit (events.js:314:20)\n    at ChildProcess.EventEmitter.emit (domain.js:483:12)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:276:12)",
    "message": "ffmpeg was killed with signal SIGKILL"
  }
}
```
</details>

I also noticed that the download dialog now just lists 2160p (the original resolution) as a download option. Anything I can do to correct that?

**Comment by violoncelloCH** [comment]
So while debugging this further I noticed that this error (at least the second one) might be coming from a low memory situation. I noticed that memory is used almost up to it's limit, so I added some more to my peertube VM after which also the encoding job for 2160p succeeded (after re-triggering it manually).
However the download dialog for the video still only shows 2160p as a download option.
